lib/kdump_utils.pm

```diff
@@ -314,7 +314,8 @@ sub check_function {
         wait_screen_change { send_key 'ret' };
     }
     elsif (is_pvm || is_ipmi) {
-        reconnect_mgmt_console;
+        # Reconnect management console on pvm only after the crash, IPMI console is managed by wait_boot
+        reconnect_mgmt_console if is_pvm;
```

Comment:
as much as possible backend specifics should be excluded from test modules. Of course, this PR does not introduce that but it just adds on top. I have read your commit message but I don't understand why PVM/IPMI needs to be handled differently. Maybe you need to change behaviour within `reconnect_mgmt_console` itself?

Reply:
`reconnect_mgmt_console` was designed for boot originally.  It contains only select_console for pvm, https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/7eb1dddaaf0244c346c9b2c17bbbdcd42cadba64/lib/utils.pm#L1418 but for IPMI it contains handling of pxeboot/ipxe logic and  boot must be confirmed https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/7eb1dddaaf0244c346c9b2c17bbbdcd42cadba64/lib/utils.pm#L1432.

IPMI console doesn't need obvious switch as it has *hidden* switch during during boot in `wait_function`.
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/7eb1dddaaf0244c346c9b2c17bbbdcd42cadba64/lib/opensusebasetest.pm#L1100

I wanted to improve `reconnect_mgmt_console` in previous PR, but imho it was adding complexity which we don't want.
If the function is designed to boot, we need to somehow let it know, that this is not boot for the call. Maybe new function `select_mgmt_console` which will contain only `select_console` in if-else for particular backend could do the job, not sure.

There is also ticket(mentioned in desc) for abuse of  `reconnect_mgmt_console if is_pvm` in test code and I made note about it there to check. This is intention to address backend specifics things (as you want), but kdump is quite specific, different hardware/scenarios needs different approach.

Reply:
yeah, I understand. Thank you for the very detailed response. This shows that you understand the general problem the same as I do :) Ok, nevermind then. Maybe you will find the time to improve the overall situation and e.g. make sure that the method `reconnect_mgmt_console` and boot handling has the same pre- and post-conditions on all backends so that the test modules calling these methods do not need to distinguish